state/machine.go

```diff
@@ -1612,6 +1612,30 @@ func (m *Machine) ProviderAddresses() (addresses corenetwork.SpaceAddresses) {
 	return
 }
 
+// ProviderAddressesBySpaceName groups the machine addresses by space name and
+// returns the result as a map where the space name is used a the key.
+func (m *Machine) ProviderAddressesBySpaceName() (map[string][]corenetwork.SpaceAddress, error) {
```

Comment:
I think this should be `AddressesBySpaceName`. These are link-layer device addresses. Those addresses might be sourced from the agent-local machiner.

Reply:
Renamed.